From 816bd950c67f114068761cb311a8185cd2c3f6dd Mon Sep 17 00:00:00 2001
From: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
Date: Mon, 1 Jun 2020 15:12:57 +0100
Subject: [PATCH 1/4] ucm2: codecs: wcd934x: Add Codec sequences.

Add WCD934x codec sequences to be used by other platforms like DB845c
and Lenovo YOGA C630

Signed-off-by: Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
---
 ucm2/codecs/wcd934x/DefaultDisableSeq.conf      |  5 +++++
 ucm2/codecs/wcd934x/DefaultEnableSeq.conf       | 14 ++++++++++++++
 ucm2/codecs/wcd934x/HeadphoneDisableSeq.conf    |  6 ++++++
 ucm2/codecs/wcd934x/HeadphoneEnableSeq.conf     |  6 ++++++
 ucm2/codecs/wcd934x/HeadphoneMicDisableSeq.conf |  2 ++
 ucm2/codecs/wcd934x/HeadphoneMicEnableSeq.conf  |  3 +++
 ucm2/codecs/wcd934x/SpeakerDisableSeq.conf      |  4 ++++
 ucm2/codecs/wcd934x/SpeakerEnableSeq.conf       |  4 ++++
 8 files changed, 44 insertions(+)
 create mode 100644 ucm2/codecs/wcd934x/DefaultDisableSeq.conf
 create mode 100644 ucm2/codecs/wcd934x/DefaultEnableSeq.conf
 create mode 100644 ucm2/codecs/wcd934x/HeadphoneDisableSeq.conf
 create mode 100644 ucm2/codecs/wcd934x/HeadphoneEnableSeq.conf
 create mode 100644 ucm2/codecs/wcd934x/HeadphoneMicDisableSeq.conf
 create mode 100644 ucm2/codecs/wcd934x/HeadphoneMicEnableSeq.conf
 create mode 100644 ucm2/codecs/wcd934x/SpeakerDisableSeq.conf
 create mode 100644 ucm2/codecs/wcd934x/SpeakerEnableSeq.conf

diff --git a/ucm2/codecs/wcd934x/DefaultDisableSeq.conf b/ucm2/codecs/wcd934x/DefaultDisableSeq.conf
new file mode 100644
index 0000000..88d82a3
--- /dev/null
+++ b/ucm2/codecs/wcd934x/DefaultDisableSeq.conf
@@ -0,0 +1,5 @@
+cset "name='RX INT7_1 MIX1 INP0' ZERO"
+cset "name='RX INT8_1 MIX1 INP0' ZERO"
+cset "name='RX INT1_2 MUX' ZERO"
+cset "name='RX INT2_2 MUX' ZERO"
+cset "name='CDC_IF TX0 MUX' ZERO"
diff --git a/ucm2/codecs/wcd934x/DefaultEnableSeq.conf b/ucm2/codecs/wcd934x/DefaultEnableSeq.conf
new file mode 100644
index 0000000..e5bdbbd
--- /dev/null
+++ b/ucm2/codecs/wcd934x/DefaultEnableSeq.conf
@@ -0,0 +1,14 @@
+cset "name='SLIM RX1 MUX' AIF1_PB"
+cset "name='SLIM RX2 MUX' AIF1_PB"
+cset "name='SLIM RX6 MUX' ZERO"
+cset "name='SLIM RX7 MUX' ZERO"
+cset "name='SLIM RX3 MUX' ZERO"
+cset "name='SLIM RX4 MUX' ZERO"
+cset "name='SLIM RX5 MUX' ZERO"
+cset "name='AIF1_CAP Mixer SLIM TX0' 1"
+
+cset "name='RX INT7_1 MIX1 INP0' RX1"
+cset "name='RX INT8_1 MIX1 INP0' RX2"
+cset "name='RX INT1_2 MUX' RX1"
+cset "name='RX INT2_2 MUX' RX2"
+cset "name='CDC_IF TX0 MUX' DEC0"
diff --git a/ucm2/codecs/wcd934x/HeadphoneDisableSeq.conf b/ucm2/codecs/wcd934x/HeadphoneDisableSeq.conf
new file mode 100644
index 0000000..0206e87
--- /dev/null
+++ b/ucm2/codecs/wcd934x/HeadphoneDisableSeq.conf
@@ -0,0 +1,6 @@
+cset "name='COMP1 Switch' 0"
+cset "name='COMP2 Switch' 0"
+cset "name='RX1 Digital Volume' 0"
+cset "name='RX2 Digital Volume' 0"
+cset "name='RX INT1 DEM MUX' ZERO"
+cset "name='RX INT2 DEM MUX' ZERO"
diff --git a/ucm2/codecs/wcd934x/HeadphoneEnableSeq.conf b/ucm2/codecs/wcd934x/HeadphoneEnableSeq.conf
new file mode 100644
index 0000000..5ce7950
--- /dev/null
+++ b/ucm2/codecs/wcd934x/HeadphoneEnableSeq.conf
@@ -0,0 +1,6 @@
+cset "name='COMP1 Switch' 1"
+cset "name='COMP2 Switch' 1"
+cset "name='RX INT1 DEM MUX' CLSH_DSM_OUT"
+cset "name='RX INT2 DEM MUX' CLSH_DSM_OUT"
+cset "name='RX1 Digital Volume' 68"
+cset "name='RX2 Digital Volume' 68"
diff --git a/ucm2/codecs/wcd934x/HeadphoneMicDisableSeq.conf b/ucm2/codecs/wcd934x/HeadphoneMicDisableSeq.conf
new file mode 100644
index 0000000..5558446
--- /dev/null
+++ b/ucm2/codecs/wcd934x/HeadphoneMicDisableSeq.conf
@@ -0,0 +1,2 @@
+cset "name='AMIC MUX0' ZERO"
+cset "name='ADC2 Volume' 0"
diff --git a/ucm2/codecs/wcd934x/HeadphoneMicEnableSeq.conf b/ucm2/codecs/wcd934x/HeadphoneMicEnableSeq.conf
new file mode 100644
index 0000000..ad43830
--- /dev/null
+++ b/ucm2/codecs/wcd934x/HeadphoneMicEnableSeq.conf
@@ -0,0 +1,3 @@
+cset "name='AMIC MUX0' ADC2"
+cset "name='ADC2 Volume' 12"
+cset "name='ADC MUX0' AMIC"
diff --git a/ucm2/codecs/wcd934x/SpeakerDisableSeq.conf b/ucm2/codecs/wcd934x/SpeakerDisableSeq.conf
new file mode 100644
index 0000000..3c977c8
--- /dev/null
+++ b/ucm2/codecs/wcd934x/SpeakerDisableSeq.conf
@@ -0,0 +1,4 @@
+cset "name='RX7 Digital Volume' 0"
+cset "name='RX8 Digital Volume' 0"
+cset "name='COMP7 Switch' 0"
+cset "name='COMP8 Switch' 0"
diff --git a/ucm2/codecs/wcd934x/SpeakerEnableSeq.conf b/ucm2/codecs/wcd934x/SpeakerEnableSeq.conf
new file mode 100644
index 0000000..2d9a3d8
--- /dev/null
+++ b/ucm2/codecs/wcd934x/SpeakerEnableSeq.conf
@@ -0,0 +1,4 @@
+cset "name='COMP7 Switch' 1"
+cset "name='COMP8 Switch' 1"
+cset "name='RX7 Digital Volume' 80"
+cset "name='RX8 Digital Volume' 80"
-- 
2.27.0.rc0

